Cilsil/Sil/Expressions/ConstExpression.cs

```diff
@@ -121,10 +120,40 @@ private ConstKind GetKindFromValue(object value)
                 default:
                     throw new ArgumentException(
                         $@"Cannot infer constant kind from value of type 
-                           {value.GetType().ToString()}");
+                           {value.GetType()}");
             }
         }
 
+        /// <summary>
+        /// Helper method for comparing values of a pair of <see cref="ConstExpression"/>.
+        /// </summary>
+        /// <param name="firstValue">The first value.</param>
+        /// <param name="secondValue">The second value.</param>
+        /// <returns><c>true</c> if the the values are equal, and <c>false</c> otherwise</returns>
+        private static bool CompareConstExpressionValue(object firstValue, object secondValue)
+        {
+            if (GetKindFromValue(firstValue) == GetKindFromValue(secondValue))
+            {
+                switch (firstValue)
+                {
+                    case IntRepresentation intRepresentation:
+                        return intRepresentation.Equals(secondValue);
+                    case ProcedureName procedureName:
+                        return procedureName.Equals(secondValue);
+                    case string _:
+                        return (string)firstValue == (string)secondValue;
+                    case float _:
+                    case double _:
+                        return (float)firstValue == (float)secondValue;
```

Comment:
comparing equality between floating point numbers could be dangerous, maybe check if they are within epsilon of each other

Reply:
Done! Thx